src/data/dataMethods.ts

```diff
@@ -547,3 +562,32 @@ function _createAbortError(signal?: AbortSignal) {
 
   return error
 }
+
+const resourceDataBase = (config: InitializedClientConfig): string => {
+  if (!config['~experimental_resource']) {
+    throw new Error('`resource` must be provided to perform resource queries')
+  }
+  const {type, id} = config['~experimental_resource']
+
+  switch (type) {
+    case 'dataset': {
+      const segments = id.split('.')
+      if (segments.length !== 2) {
+        throw new Error('Dataset ID must be in the format "project.dataset"')
```

Comment:
And here?

Reply:
In theory should never reach this point as we have the same check when setting the config. We have:
```
test('throws if resource type is dataset and id has no dots', () => {
      expect(() =>
        createClient({'~experimental_resource': {type: 'dataset', id: 'abc123'}}),
      ).toThrow(/Dataset resource ID must be in the format "project.dataset"/)
    })```